### ISSUE TITLE
Visulization of the image batch

### ISSUE THREAD

**Issue by gaopeng-eugene** [DrSleep/tensorflow-deeplab-resnet#8]
Hi, I use the matplotlib to visualise the image batch for debug purpose.
However, for color image, the maplotlib can only show noise like this

<img width="268" alt="a08c3fc2-8ecd-413f-8bc8-eb2ac810923b" src="https://cloud.githubusercontent.com/assets/8071803/22009499/a00d0f6a-dcbe-11e6-9e13-b68db399bddd.png">

The single channel is correct
<img width="293" alt="8875bfb4-9f2e-47cd-887c-21ac574e943b" src="https://cloud.githubusercontent.com/assets/8071803/22009552/f20407ec-dcbe-11e6-941d-223092d5597e.png">

**Comment by DrSleep** [comment]
Hi, that's weird.
Did you use the provided visualisation code or did you write your own?
Because in my case, the generated visualisations look like this, and it is a colour image input:
![vis](https://cloud.githubusercontent.com/assets/7841432/22010389/440f35d6-dccc-11e6-84fb-1799a5a91b4a.png)

Besides, yesterday I created a new branch with visualisation embedded in tensorboard, you might want to check it out as well - https://github.com/DrSleep/tensorflow-deeplab-resnet/tree/image-summary

**Comment by gaopeng-eugene** [comment]
I will try soon. mask and pred works quite well except the data part.

**Comment by gaopeng-eugene** [comment]
I use my own visulization code

**Comment by DrSleep** [comment]
When you visualise an image from the image batch, you should apply the inverse pre-processing transformation: i.e., add the mean vector and convert from BGR to RGB. 
This is the line that does exactly that: https://github.com/DrSleep/tensorflow-deeplab-resnet/blob/master/train.py#L167

**Comment by gaopeng-eugene** [comment]
Here is my code for network input image visualization

import argparse
from datetime import datetime
import os
import sys
import time

import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt

import tensorflow as tf
import numpy as np

from deeplab_resnet import DeepLabResNetModel, ImageReader, decode_labels, prepare_label

input_size = (321,321)
data_dir = '/opt/data/penggao/dataset/VOCdevkit/VOC2012_aug/VOC2012'
data_list = './dataset/train.txt'
RANDOM_SCALE = True
coord = tf.train.Coordinator()
reader = ImageReader(
            data_dir,
            data_list,
            input_size,
            RANDOM_SCALE,
            coord)
image_batch, label_batch = reader.dequeue(4)

config = tf.ConfigProto()
config.gpu_options.allow_growth = True
sess = tf.Session(config=config)
init = tf.initialize_all_variables()
sess.run(init)

threads = tf.train.start_queue_runners(coord=coord, sess=sess)

images = sess.run(image_batch)

import matplotlib.pyplot as plt
%matplotlib inline  
plt.imshow(images[0])

**Comment by gaopeng-eugene** [comment]
I add the mean vector and restore the original RGB code, but it didn't work either. By the way, the paste code didn't contain the restore implementation.

**Comment by DrSleep** [comment]
Did you convert to uint8?

**Comment by gaopeng-eugene** [comment]
convert to uint8 didn't work either

**Comment by DrSleep** [comment]
I cannot reproduce your error; it is working in my case: https://gist.github.com/DrSleep/d09d256d301569bebf4864df8b999602#file-image-ipynb

**Comment by gaopeng-eugene** [comment]
Problem solved. convert to uint8 can solve the problem
